examples/jsm/lines/LineSegments2.js

```diff
@@ -147,135 +232,212 @@ class LineSegments2 extends Mesh {
 		const distanceToBox = Math.max( camera.near, _box.distanceToPoint( ray.origin ) );
 
 		// increase the box bounds by the worst case line screen space width
-		const boxMargin = getWorldSpaceHalfWidth( camera, distanceToBox, lineWidth, resolution );
-		_box.max.x += boxMargin;
-		_box.max.y += boxMargin;
-		_box.max.z += boxMargin;
-		_box.min.x -= boxMargin;
-		_box.min.y -= boxMargin;
-		_box.min.z -= boxMargin;
+		let boxMargin;
+		if ( this.material.worldUnits ) {
 
-		if ( raycaster.ray.intersectsBox( _box ) === false ) {
+			boxMargin = lineWidth * 0.5;
 
-			return;
+		} else {
+
+			boxMargin = getWorldSpaceHalfWidth( camera, distanceToBox, lineWidth, resolution );
 
 		}
 
-		//
+		_box.expandByScalar( boxMargin );
 
-		// pick a point 1 unit out along the ray to avoid the ray origin
-		// sitting at the camera origin which will cause "w" to be 0 when
-		// applying the projection matrix.
-		ray.at( 1, _ssOrigin );
+		if ( raycaster.ray.intersectsBox( _box ) === false ) {
 
-		// ndc space [ - 1.0, 1.0 ]
-		_ssOrigin.w = 1;
-		_ssOrigin.applyMatrix4( camera.matrixWorldInverse );
-		_ssOrigin.applyMatrix4( projectionMatrix );
-		_ssOrigin.multiplyScalar( 1 / _ssOrigin.w );
+			return;
 
-		// screen space
-		_ssOrigin.x *= resolution.x / 2;
-		_ssOrigin.y *= resolution.y / 2;
-		_ssOrigin.z = 0;
+		}
 
-		_ssOrigin3.copy( _ssOrigin );
+		//
 
+		let faceIndex;
+		let point;
+		let pointOnLine;
+		let dist = Infinity;
+		let isLoopingInside = false; // is true while isInside == true
 		_mvMatrix.multiplyMatrices( camera.matrixWorldInverse, matrixWorld );
 
-		for ( let i = 0, l = instanceStart.count; i < l; i ++ ) {
+		if ( worldUnits ) {
 
-			_start4.fromBufferAttribute( instanceStart, i );
-			_end4.fromBufferAttribute( instanceEnd, i );
+			for ( let i = 0, l = instanceStart.count; i < l; i ++ ) {
 
-			_start4.w = 1;
-			_end4.w = 1;
+				computeStartEnd( instanceStart, instanceEnd, i );
 
-			// camera space
-			_start4.applyMatrix4( _mvMatrix );
-			_end4.applyMatrix4( _mvMatrix );
+				// skip the segment if it's entirely behind the camera
+				const isBehindCameraNear = _start4.z > near && _end4.z > near;
+				if ( isBehindCameraNear ) {
 
-			// skip the segment if it's entirely behind the camera
-			const isBehindCameraNear = _start4.z > near && _end4.z > near;
-			if ( isBehindCameraNear ) {
+					continue;
 
-				continue;
+				}
 
-			}
+				// trim the segment if it extends behind camera near
+				trimSegment( near );
 
-			// trim the segment if it extends behind camera near
-			if ( _start4.z > near ) {
+				if ( ! isInClipSpace( projectionMatrix, resolution ) ) {
 
-				const deltaDist = _start4.z - _end4.z;
-				const t = ( _start4.z - near ) / deltaDist;
-				_start4.lerp( _end4, t );
+					continue;
 
-			} else if ( _end4.z > near ) {
+				}
```

Comment:
For the world units lines there's no reason to ensure the lines are in clip space anymore. The only reason the camera is required otherwise is because it's necessary they are in screen space. I think the simplest approach would be to iterate over the lines without any screenspace or camera logic and checking ray / line segment distances for intersections. Then there's no requirement that the lines be visible in order to be interacted with. Let me know if that sounds good to you!

Reply:
I'm not sure I understand - let's say we have a very long line that's going from behind the camera near to inside the camera frustum
We don't want the parts of the lines before the camera near to be considered, but if we only keep the intersections and try to calculate it at the end we will be missing some information - no?

I agree about the clipSpace, remove the `if` block for that
Let me know if you meant to remove the trimSegment and isBehindCameraNear as well

Reply:
Also, out of curiosity - why don't we need to consider far?
The raytracer removes points that are behind the far plane - but sometimes a closer part of the line might fall within the threshold so we would miss intersections there, am I missing something?

Reply:
There's no reason that world lines have to be handled in screen space. The only reason the screen space lines used the camera is because it's required based on how they're displayed. Picking effectively happens in 2D and given the way they're rendering proper picking is effectively undefined outside of the camera clip space. It's definitely a special case for the screen space line variant.

This isn't the case with the world unit lines, though. They're basically rendered as capsules. Meshes, for example, don't have special handling when they're behind the camera and don't consider the camera at all. This enables rays that are not derived from a camera and mouse coordinate to be used for raycasting (ie vr controller rays outside of view or rays for game mechanics outside of view)

The built in [GL line raycasting](https://github.com/mrdoob/three.js/blob/1c553914ace73127cc67660597806b2c7c013548/src/objects/Line.js#L160-L189) doesn't use a camera and includes support for a threshold which effectively means they're treated as capsules. I think that's more along the lines of what I imagine for world unit raycasting.

Reply:
That would mean the "computeStartEnd", "trimSegment", and "isInClipSpace" functions do not need to be shared and don't need to be pulled out into separate functions. I just want to keep the changes to the original logic minimal.

Reply:
Oh I forgot to update - I changed the main camera's near to `10` to test behavior and luckily committed it by accident so it can still be tested:


https://user-images.githubusercontent.com/37291459/154812175-afe783da-5fa0-49ec-813a-5c5e8ca22ac8.mov


The lines are rendered truncated by the near plane and the intersection still happens (denoted by the cursor and the little canvas on the bottom left which still has near set to `1` - which I'm pretty sure is a bug in my last commits
I just want to confirm - this is the desired behavior? if so I'll remove the calls to trimSegment/isBehindCamera instead of debugging

Reply:
I believe this is a function of the math involved. The lines are rendered as capsules and have thickness -- you're not seeing back faces of the capsule but technically the near plane (and therefore the ray start) is within the capsule and registers as a "hit" within the given threshold.